### PR TITLE
New branch edit tags

### PR DESCRIPTION
**Pull request by wang-xinrong**
#4 New branch for edit tags

### CODE REVIEW

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 1]
```diff
@@ -55,7 +56,7 @@ public static ReadOnlyAddressBook getSampleAddressBook() {
      */
     public static Set<Tag> getTagSet(String... strings) {
         return Arrays.stream(strings)
-                .map(Tag::new)
+                .map(tagName -> new Tag(tagName, TagStatus.DEFAULT_STATUS))
```
Comment (yyccbb) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:59]: Can be modified in the future.

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 2]
```diff
@@ -115,6 +116,14 @@ public PersonBuilder withEmail(String email) {
...
     }
 
+    /**
+     * Sets the {@code Tag} of the {@code Person} that we are building.
+     */
+    public PersonBuilder withTag(String tagName, TagStatus tagStatus) {
+        this.tags = Tag.updateTagsWithNewTag(new HashSet<>(tags), tagName, tagStatus);
+        return this;
+    }
+
```
Comment (yyccbb) [inline src/test/java/seedu/address/testutil/PersonBuilder.java:126]: The current name is a bit confusing, can change to something like "updateTag'.
Also it is creating a hashset every time so a bit inefficient (but can ignore for now).

**Review by yyccbb — APPROVED** [review]
Some minor changes are needed, but sufficiently good to be merged.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -154,4 +156,13 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
...
     }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    public static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+
```
[the remark this answers is not in the document]
Comment (wang-xinrong) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:167]: hmm, I think this method was originally here though, was it placed somewhere else previously?

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 4]
```diff
@@ -56,7 +69,29 @@ public int hashCode() {
      * Format state as text for viewing.
      */
     public String toString() {
-        return '[' + tagName + ']';
+        return "[ " + tagName + " : " + tagStatus + " ]";
+    }
+
+    public static void isTagNameValid(String tagName) throws IllegalArgumentException {
+        checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
+    }
```
[the remark this answers is not in the document]
Comment (wang-xinrong) [reply src/main/java/seedu/address/model/tag/Tag.java:77]: Sure, I was following the original implementation but we can do updates in the future.

**Pull request merged**